palace.pyx

```diff
@@ -1158,6 +1176,116 @@ cdef class SourceGroup:
         """
         self.impl.destroy()
 
+
+cdef class AuxiliaryEffectSlot:
+    """An effect processor.
+
+    It takes the output mix of zero or more sources,
+    applies DSP for the desired effect (as configured
+    by a given `Effect` object), then adds to the output mix.
+
+    Parameters
+    ----------
+    context : Context
+        The context from which the auxiliary effect slot is to be created.
+
+    Raises
+    ----------
+    RuntimeError
+        If the effect slot can't be created.
+    """
+    cdef alure.AuxiliaryEffectSlot impl
+
+    def __init__(self, context: Context) -> None:
+        self.impl = (<Context> context).impl.create_auxiliary_effect_slot()
+
+    def __enter__(self) -> AuxiliaryEffectSlot:
+        return self
+
+    def __exit__(self, exc_type: Optional[Type[BaseException]],
+                 exc_val: Optional[BaseException],
+                 exc_tb: Optional[TracebackType]) -> Optional[bool]:
+        self.destroy()
+
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl < (<AuxiliaryEffectSlot> other).impl
+
+    def __le__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl <= (<AuxiliaryEffectSlot> other).impl
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl == (<AuxiliaryEffectSlot> other).impl
+
+    def __ne__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl != (<AuxiliaryEffectSlot> other).impl
+
+    def __gt__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl > (<AuxiliaryEffectSlot> other).impl
+
+    def __ge__(self, other: Any) -> bool:
+        if not isinstance(other, AuxiliaryEffectSlot):
+            return NotImplemented
+        return self.impl >= (<AuxiliaryEffectSlot> other).impl
+
+    def __bool__(self) -> bool:
+        return <boolean> self.impl
+
+    def set_gain(self, gain: float) -> None:
+        return self.impl.set_gain(gain)
+
+    def set_send_auto(self, sendauto: bool) -> None:
+        return self.impl.set_send_auto(sendauto)
+
+    gain = property(fset=set_gain, doc=(
+        """If set to true, the reverb effect will automatically
+        apply adjustments to the source's send slot gains based
+        on the effect properties.
+
+        Has no effect when using non-reverb effects. Default is true.
+        """
+    ))
+    send_auto = property(fset=set_send_auto, doc= (
+        """Update the effect slot with a new effect.
+        The given effect object may be altered or
+        destroyed without affecting the effect slot.
+        """
+    ))
```

Comment:
This is much better, but try to not leave closing parentheses alone like this (sorry I have Lisp background).